### DISCUSSION TITLE
ODF sections in subplots

### DISCUSSION THREAD

**Discussion by Gbowker** [mtex-toolbox/mtex#1231] [Ask Anything] [answered]
Hello all,
I'm trying to compare ODF sections of different BCC textures, and would like to plot them as part of the same subplot to scale them equally. This works fine with pole-plots, (using plotPDF) but trying to produce a new ODF section in a new subplot position overwrites the plot in the first position. Anybody else having this issue? Please see below for the code and figures in question. Thank you, **any** help is appreciated.

### plot_BCC_suplots.m
```Matlab
fig_array = [2,4]; % define subplot
newMtexFigure('layout', fig_array, 'figSize','huge'); % tell mtex about subplot

ODFs_to_plot = ["ODF1", "ODF2", "etc."]
for index = 1:length(ODFs_to_plot)
    % code to calculate odf from quaternion data e.g. get_odf(ODFs_to_plot{index})

    figure(1);
    subplot(fig_array(1), fig_array(2), index);
    BCC_odfsection(odf, 45)
end

CLim(gcm,'equal');
mtexColorbar('title', 'mrd', 'location', 'eastoutside', 'FontSize', 52);
```

### BCC_odfsection.m
```Matlab
function [BCC_odfsection] = BCC_odfsection(odf, slice)
    odf.SS=specimenSymmetry('orthorhombic');
    plotSection(odf,'contourf','phi2',slice*degree, 'minmax');
    [rgb] = add_white(viridis,10); % make mrd=0 white
    colormap(rgb);
    f = gcm;
    for i=1:length(f.children)
     sp = getappdata(f.children(i),'sphericalPlot');
     sp.TR.Position = [1.15, 1, 1];               % slice label position
     sp.TL.Position=[1.05, 0.5, 1];               % max label posn
     sp.BL.Position=[1.05, -0.01, 1];              % min label posn
    end
end
```

### example pole figure subplot:
![pole_figure example](https://github.com/Gbowker/project_scripts/blob/master/00r.jpeg)
I would like my ODF section figure to look like this.

### example ODF section:
![pole_figure example](https://github.com/Gbowker/project_scripts/blob/master/beta_50r.jpeg)

**Comment by kilir** [comment, marked as answer]
Hi,
is there any particular reason `nextAxis()` doesn't work?
Cheers,
Rüdiger

**Reply by Gbowker** [reply]
This works but how can I change the dimensions of my subplot? For example have 2 rows, 4 columns instead of 2 rows, 3 columns?

**Reply by kilir** [reply]
Either specify `'nrows'` and `'ncols'` or `'layout'` while initiating mtexFigure or since this commit e45865d9c29a578fed77bb7b26888fa6a03cf19b you can change it via `nextAxis(i,j)`.
Hope that helps.

Cheers,
Rüdiger

**Reply by Gbowker** [reply]
This helps a lot! Thank you Rüdiger!

**Reply by JOchoaA** [reply]
Hello all,

I am trying to do the same, but I need each ODF plot with a different color map. I change the colormap depending on the files' name; I do so directly on the plotting function. 

As a result, when I plot every ODF individual, the colormap is different for all of them, but when I try to make the subplot, they all have the same color.

could you guys please help me?

I appreciate your time and any help.

Individual ones

![alamel_odf](https://user-images.githubusercontent.com/61281507/217513720-7993a8e7-7525-49dd-a36a-57db7643cdc2.png)
![fct_odf](https://user-images.githubusercontent.com/61281507/217513724-ef95e36a-1511-4e3f-bab8-e1af87320ce6.png)
![vpsc_odf](https://user-images.githubusercontent.com/61281507/217513726-cd72eb20-5ab4-49ad-bbd9-9b0be024bae2.png)

All together

![p20_odfs](https://user-images.githubusercontent.com/61281507/217513305-bd546607-dde6-4239-bede-69ae02480d9e.png)

**Reply by kilir** [reply]
Hi,
```
cs = crystalSymmetry('432');

plotSection(unimodalODF(orientation.rand(3,cs)),'sections',4)

f=gcm;
cm = {'winter', 'spring', 'summer', 'autumn'}
% for three different colormaps 
for i=1:length(f.children)
  mtexColorMap(f.children(i),char(cm(i)))
end
```

Cheers,
Rüdiger

**Reply by JOchoaA** [reply]
Hi Rüdiger,

Thanks a lot; now it is working perfectly.

Best wishes,

John.
